dds/idl/dynamic_data_adapter_generator.cpp

```diff
@@ -450,26 +454,133 @@ namespace {
           "\n";
       }
 
+      be_global->header_ <<
+        "  DDS::DynamicData_ptr clone();\n"
+        "  bool serialized_size(const OpenDDS::DCPS::Encoding& enc, size_t& size, OpenDDS::DCPS::Sample::Extent ext) const;\n"
+        "  bool serialize(OpenDDS::DCPS::Serializer& ser, OpenDDS::DCPS::Sample::Extent ext) const;\n"
+        "\n";
+
+      const bool is_topic_type = be_global->is_topic_type(node);
```

Comment:
A lot of the code below is can be simplified by making full use of `RefWrapper`. Here's one example:

https://github.com/OpenDDS/OpenDDS/blob/4d1d6a37ed7d86978600edf631e7bb9f1772a5aa/dds/idl/marshal_generator.cpp#L1607-L1612

It will make some decisions based on the type and others based on traits set before calling `done`.

Reply:
`RefWrapper` is used for types that need `IDL::DistinctType`. The names with `_forany` suffix are manually constructed because `RefWrapper` will generate references (I think this is needed by marshal_generator) while I need to get non-reference types to instantiate values for them. I think it's simple enough to construct those type names manually without needing to change `RefWrapper`.